### DISCUSSION TITLE
Three questions about tags

### DISCUSSION THREAD

**Discussion by TigBuh99** [keepassxreboot/keepassxc#9198] [Q&A] [answered]
I wonder if there is a way to do a search for tags? Say I want to search for the tags "apples" AND "oranges". Or tags "bananas" OR "pears".

What if I want to search for "cherries" as a word somewhere in my database, but **not** the tag "cherries"? Or the opposite of this.

Also what if I would like to see all entries that **do not** have any tags?

**Comment by droidmonkey** [comment, marked as answer]
Here are some searches you can try:

1. Find entries with both tags: `tag:apples tag:oranges`

2. Find entries with either tag: `*tag:bananas|pears`

3. Just type in cherries to search the word, tags are not part of the default search fields. You can also explicitly exclude entries with the cherries tag using: `!tag:cherries`

4. To find entries without tags: `!tag:*`

More information: https://keepassxc.org/docs/KeePassXC_UserGuide.html#_searching_the_database

**Comment by TigBuh99** [comment]
Thank you for the hints. I tried them out on my database, but 1 and 2 don't give me the right answers. Number 3 works correctly.

Pity about number 4. I have to go through a large number of entries in order to find which ones don't have a tag in order to adjust them as I like. Well, at least I'll have to do it only once!

**Reply by droidmonkey** [reply]
Note that tags are case sensitive

**Reply by droidmonkey** [reply]
They are also exact match by default so `apple` will NOT match `apples` or `Apple`

**Comment by TigBuh99** [comment]
Thank you for the additional information. The only thing that I can think of is that one of my real case tags consists of two words. I put it in double quotes, like _tags:"Fruit salad"_. I don't know if this makes a difference as far as the first case is concerned when both tags should exist for the result to be true. In my actual database I have 23 entries with the tags _Fruit salad_ and _apple_ and three entries with the tag _Fruit salad_ only. My query _tags:apple tags:"Fruit salad"_ returns 26 results. Shouldn't I be seeing only 23 results for the query to be correct?

Please note that if I search for the two words tag say, _tags:"Fruit salad"_ I get 26 results which is correct.

**Reply by droidmonkey** [reply]
I cannot replicate your findings:

![image](https://user-images.githubusercontent.com/2809491/222928910-5994e315-82c5-430f-8463-edbded5e773b.png)

![image](https://user-images.githubusercontent.com/2809491/222928915-46a207a0-a363-4d1f-a102-f3c223361dc2.png)

**Reply by droidmonkey** [reply]
Also I figured out your no-tags search, use `!tag:*`

**Reply by TigBuh99** [reply]
> Also I figured out your no-tags search, use `!tag:*`

This is perfect for me! I have about 180 entries out of nearly 300 with no tags. This search term makes it very easy to trace them.

**Reply by TigBuh99** [reply]
> I cannot replicate your findings:
Could you try it where one of the entries is tagged with a two words tag?
I uploaded an example. The password is 1234.
[ForTesting.zip](https://github.com/keepassxreboot/keepassxc/files/10890896/ForTesting.zip)
I cannot get the OR term to work as for example *tag:orange|"Fruit salad". It returns no result but I expected to see Apples and Oranges.

By the way I found a mistake that I was making. There must be no space on either side of the vertical bar. I write this in case some other newbie makes the same mistake.

EDIT: Quick check. If you change the tag from *Fruit salad* to *Fruit-salad* and use it without the quotes, then the search *tag:orange|Fruit-salad works as would be expected.
